folly/container/detail/F14Table.h

```diff
@@ -187,6 +187,15 @@ struct StdNodeReplica<
 
 #endif
 
+const auto erase_if_impl = [](auto& c, auto predicate) {
```

Comment:
Is there a reason to make this a callable global object v.s. a function template?

Reply:
No particular reason at all. In general, I reach for common lambdas over function templates as it's less verbose syntax due to `auto` parameters rather than them being deduced from templates.

Should I prefer function templates over lambdas for common helper functions in this codebase/here in general? I don't have strong opinions, so I'm happy to adopt whichever is preferred.

Reply:
If there is a specific reason that a lambda here is better than a function template then go with the lambda, but otherwise function templates are the norm in folly.

Reply:
Just switched to a function template for `erase_if_impl` to stay consistent with other parts of the code. The lambda doesn't buy us much here.